### PR TITLE
Update docs to show more descriptions, links, and features

### PR DESCRIPTION
**Pull request by lmd59**
# Summary

- Makes updates to the main readme to link to the sub-readmes and show a diagram describing how they are related.
- Updates the app readme to provide documentation on the frontend features.

## New behavior
No new functionality. Updates documentation.

## Code changes
None. Documentation only.

# Testing guidance
Review the updates readmes: [main](https://github.com/projecttacoma/measure-repository/blob/update-docs/README.md) and [app](https://github.com/projecttacoma/measure-repository/blob/update-docs/app/README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Measure Repository
 
-A prototype implementation of a [FHIR Measure Repository Service](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html) and associated frontend application.
+A prototype implementation of a [FHIR Measure Repository Service](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html) and associated frontend application. This repository is a monorepo that consists of: 
+- [Measure Repository Service](https://github.com/projecttacoma/measure-repository/blob/main/service/README.md)
+  - Implements portions of the [FHIR Measure Repository Service](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html) specification
+  - Acts as a FHIR server and shared source of truth for measures
```
Comment (elsaperelli) [inline README.md:6]: Is it a source of truth for just measures or measures and libraries? Or should we say artifacts?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+- [Measure Repository App](https://github.com/projecttacoma/measure-repository/blob/main/app/README.md)
+  - A prototype [Next.js](https://nextjs.org/) application that demonstrates potential interactions with a FHIR Measure Repository Service
+  - Includes a frontend, backend, and small database which acts as in-progress storage space for drafting measures
```
Comment (elsaperelli) [inline README.md:9]: I would change "drafting measures" to "draft measures and libraries" or "draft artifacts" since it holds both

File: app/README.md [hunk 3]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+
+# Frontend Features
+
+## Navigation
+
+The application is divided into two main sets of functionality, _Repository_ and _Authoring_, both of which are accessible from tabs at the top of the page.
+
+<img src="./static/tabs.png" alt="Screenshot of tabs" width="200"/>
+
+The Repository tab provides functionality for access and interacting with measure artifacts that are in the FHIR Measure Repository Service. The Authoring tab focuses on measure artifacts that are in the process of being drafted. Artifacts can be copied or moved between these two spaces through _draft_ and _release_ actions, described in the [Viewing](#viewing) and [Editing](#editing) sections.
```
Comment (elsaperelli) [inline app/README.md:55]:
```suggestion
The Repository tab provides functionality for accessing and interacting with measure artifacts that are in the FHIR Measure Repository Service. The Authoring tab focuses on measure artifacts that are in the process of being drafted. Artifacts can be copied or moved between these two spaces through _draft_ and _release_ actions, described in the [Viewing](#viewing) and [Editing](#editing) sections.
```
I think?

Comment (elsaperelli) [inline app/README.md:55]:
```suggestion
The Repository tab provides functionality for access and interacting with measure and library artifacts that are in the FHIR Measure Repository Service. The Authoring tab focuses on measure and library artifacts that are in the process of being drafted. Artifacts can be copied or moved between these two spaces through _draft_ and _release_ actions, described in the [Viewing](#viewing) and [Editing](#editing) sections.
```
I would potentially rephrase "Artifacts can be copied or moved between..." because technically they are not being copied/moved between the two because their version/status changes between the two- is this too nitpicky? You could instead keep it simple to something like "Artifacts can be drafted or released beween these two spaces..." 

I would also maybe reword "The Authoring tab focuses on measure artifacts that are in the process of being drafted" since they are drafts...maybe just say "focuses on draft measure and library artifacts" or something else that is perhaps clearer on the fact that they are drafts.

File: app/README.md [hunk 4]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+
+Within these tabs, the left panel navigation allows for selecting artifact types or searching (Repository only) for artifacts so that they can be browsed for viewing or actions.
+
+<img src="./static/navigation.png" alt="Screenshot of navigation" width="250"/>
+
+## Landing Page
+The landing page can be accessed on first load or by clicking the Measure Repository home button on the top left. This page links to resources to learn more about the Measure Repository and shows existing FHIR service capabilities.
```
Comment (elsaperelli) [inline app/README.md:62]:
```suggestion
The landing page can be accessed on first load or by clicking the Measure Repository home button on the top left. This page links to resources to learn more about the Measure Repository and shows existing FHIR service capabilities.
```
"In the top left" or "on the top left of the page"?

File: app/README.md [hunk 5]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+
+## Repository Tab
+The Repository tab displays sets of artifacts from the FHIR Measure Repository Service. Options allow for viewing resource details or reviewing the resource.
+
+<img src="./static/repository-options.png" alt="Screenshot of repository options" width="75"/>
+
+### Viewing
+-	Measure: Resource details include options to see the JSON, Data Requirements, or Dependencies.        
+-	Library: Resource details include options to see the JSON or Narrative.  
+-	Drafting: Both resource views allow the user to create a draft from the artifact. This action copies the existing artifact and any children it owns to the draft Authoring space with an incremented version to logically differentiate it from the original artifact.
```
Comment (elsaperelli) [inline app/README.md:72]:
```suggestion
-	Drafting: Both resource views allow the user to create a draft from the artifact. This action copies the existing artifact and any children it owns to the draft Authoring space with an incremented version and draft status to logically differentiate it from the original artifact.
```
Or something along those lines if you think that's necessary

File: app/README.md [hunk 6]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+
+### Reviewing
+Reviewing a resource provides options for viewing existing comments on the resource or adding a new comment. (Note: this is currently partially implemented and requires updates for viewing comments.)
+
+## Authoring Tab
+The Authoring tab gives options for creating a new draft artifact from scratch or starting from a copy of an existing artifact from the FHIR Measure Repository. 
+
+<img src="./static/draft-artifact.png" alt="Screenshot of create draft artifact" width="350"/>
+
+The left navigation resource type selection displays sets of artifacts that are in progress, being drafted for potential addition to the FHIR Measure Repository. Options allow for editing, reviewing, or deleting a draft resource.
```
Comment (elsaperelli) [inline app/README.md:82]:
```suggestion
The left navigation resource type selection displays sets of artifacts that are in draft status. Options allow for editing, reviewing, or deleting a draft resource.
```
I am not totally sure what's best here I will leave it up to you but I think that sentence could be slightly reworded?

File: app/README.md [hunk 7]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+### Editing
+-	Updating: A Measure or Library allows changes to basic resource fields:
+    - url
+    - identifier value
+    - identifier system
+    - name
+    - title
+    - description
+    - library (Measure only)
+- Releasing: Both resource types may be released from the editing page. This action publishes the draft artifact and any children it owns to the Repository space, adding them to the FHIR Measure Repository and giving them an active status. The draft artifact(s) will be removed from the Authoring space.
```
Comment (elsaperelli) [inline app/README.md:95]: Do we want this to be its own section?

File: app/README.md [hunk 8]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+The landing page can be accessed on first load or by clicking the Measure Repository home button on the top left. This page links to resources to learn more about the Measure Repository and shows existing FHIR service capabilities.
+
+## Repository Tab
+The Repository tab displays sets of artifacts from the FHIR Measure Repository Service. Options allow for viewing resource details or reviewing the resource.
+
+<img src="./static/repository-options.png" alt="Screenshot of repository options" width="75"/>
+
+### Viewing
+-	Measure: Resource details include options to see the JSON, Data Requirements, or Dependencies.        
+-	Library: Resource details include options to see the JSON or Narrative.
```
Comment (elsaperelli) [inline app/README.md:71]:
```suggestion
-	Library: Resource details include options to see the JSON and the Narrative, CQL, ELM, Data Requirements, and Dependencies if they exist.
```

File: app/README.md [hunk 9]
```diff
@@ -37,3 +42,60 @@ npm run start:app
...
+## Landing Page
+The landing page can be accessed on first load or by clicking the Measure Repository home button on the top left. This page links to resources to learn more about the Measure Repository and shows existing FHIR service capabilities.
+
+## Repository Tab
+The Repository tab displays sets of artifacts from the FHIR Measure Repository Service. Options allow for viewing resource details or reviewing the resource.
+
+<img src="./static/repository-options.png" alt="Screenshot of repository options" width="75"/>
+
+### Viewing
+-	Measure: Resource details include options to see the JSON, Data Requirements, or Dependencies.
```
Comment (elsaperelli) [inline app/README.md:70]:
```suggestion
-	Measure: Resource details include options to see the JSON and Data Requirements and Dependencies if they exist.   
```

Comment (elsaperelli) [reply app/README.md:70]: Doesn't have to be exactly that, but something that indicates that the Data Requirements and Dependencies only display if they exist (for example, if we create a draft artifact from scratch and then release it, it will not have data requirements or dependencies).

File: app/README.md (hunk 8 above)
Comment (elsaperelli) [reply app/README.md:71]: Doesn't have to be exactly that, but something that indicates that the Data Requirements, Dependencies, ELM, and CQL only display if they exist (for example, the FHIR library only has JSON and Narrative but MATGlobalCommonFunctionsFHIR4 has JSON, DR, Dependencies, CQL and ELM but no Narrative).

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Great changes and thank you for doing this much needed task!

I just have small comments but overall I agree with the structure. My only other suggestion is maybe linking to some of the specific operation definitions in the MRS spec (like `draft` and `release`).

File: README.md (hunk 1 above)
Comment (lmd59) [reply README.md:6]: Hmm- the spec says "to support authoring, publishing, and distribution of FHIR-based quality measure specifications" ... maybe we say:
```suggestion
  - Acts as a FHIR server and shared source of truth for measures and associated artifacts
```
?

File: README.md (hunk 2 above)
Comment (lmd59) [reply README.md:9]: similarly, I think the libraries are in support of the measures, so maybe
```suggestion
  - Includes a frontend, backend, and small database which acts as in-progress storage space for drafting measures and associated artifacts
```

File: app/README.md (hunk 3 above)
Comment (lmd59) [reply app/README.md:55]: So the problem is that there can technically be drafts in the FHIR repo as well, so the app is kind of a temporary space for ... lets maybe say "unofficial" draft measures (because they're not in the official repo yet). I'm trying to differentiate the purpose of the authoring space as being more temporary and in progress.

File: README.md (hunk 2 above)
Comment (lmd59) [reply README.md:9]: I'm trying to leave it open ended in case we tackle measure reports or something... heh

File: app/README.md (hunk 3 above)
Comment (lmd59) [reply app/README.md:55]:
```suggestion
The Repository tab provides functionality for access and interacting with artifacts that are in the FHIR Measure Repository Service. The Authoring tab focuses on artifacts that are in the process of being drafted. Artifact content can be copied or moved between these two spaces through _draft_ and _release_ actions, described in the [Viewing](#viewing) and [Editing](#editing) sections.
```
What do you think about changing it to "Artifact content" so that we can show that the main content is being copied or moved, but this implies that there may be some more meta-data type fields that may change in that process?

File: app/README.md (hunk 4 above)
Comment (lmd59) [reply app/README.md:62]: on the left... at the top... prepositions are hard lol
```suggestion
The landing page can be accessed on first load or by clicking the Measure Repository home button in the top left corner. This page links to resources to learn more about the Measure Repository and shows existing FHIR service capabilities.
```

File: app/README.md (hunk 6 above)
Comment (lmd59) [reply app/README.md:82]: The app really sits in a bit of a weird space - I wanted to make clear that the draft measures are not in the FHIR repository but rather just in our local storage until they are "released" (or otherwise moved to the official FHIR repo).

File: app/README.md (hunk 7 above)
Comment (lmd59) [reply app/README.md:95]: I've kind of organized everything along the current navigation hierarchy. Releasing is a sub-function only available once you start editing, so that's why it's in the editing section. Could potentially reorganize if there's a different organizational scheme that makes more sense!

File: app/README.md (hunk 9 above)
Comment (lmd59) [reply app/README.md:70]: Similar to the library comment on the line below, did I miss options for Measure? Like I think measure could have narrative too? Could it have elm or cql?

File: app/README.md (hunk 8 above)
Comment (lmd59) [reply app/README.md:71]: Good catch!
```suggestion
-	Library: Resource details show the JSON as well as options to see the ELM, CQL, Narrative, Data Requirements, and Dependencies, if they exist.  
```

File: app/README.md (hunk 9 above)
Comment (lmd59) [reply app/README.md:70]: If not, then it would probably just be
```suggestion
-	Measure: Resource details show the JSON as well as options to see the Narrative, Data Requirements, and Dependencies, if they exist.          
```

File: app/README.md (hunk 5 above)
Comment (lmd59) [reply app/README.md:72]:
```suggestion
-	Drafting: Both resource views allow the user to create a draft from the artifact. This action copies the existing artifact and any children it owns to the draft Authoring space with a draft status and an incremented version to logically differentiate it from the original artifact. See the [draft operation specification](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#draft).
```

File: app/README.md (hunk 7 above)
Comment (lmd59) [reply app/README.md:95]:
```suggestion
- Releasing: Both resource types may be released from the editing page. This action publishes the draft artifact and any children it owns to the Repository space, adding them to the FHIR Measure Repository and giving them an active status. The draft artifact(s) will be removed from the Authoring space. See the [release operation specification](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#release).
```

File: app/README.md (hunk 6 above)
Comment (elsaperelli) [reply app/README.md:82]: Ah I see what you mean. I think you can keep essentially what you have but should there be an "and" instead of the comma? Idk grammar is hard

Comment (lmd59) [reply app/README.md:82]: agreed on the "and"!
```suggestion
The left navigation resource type selection displays sets of artifacts that are in progress and being drafted for potential addition to the FHIR Measure Repository. Options allow for editing, reviewing, or deleting a draft resource.
```

**Review by elsaperelli — APPROVED** [review]
Good changes! I am not terribly worried about it so I will leave it up to you but I am still a bit weary on the "measures and their associated artifacts" wording. I agree with your reasoning of the spec saying "measure specifications" (and it IS the _measure_ repository service after all...), but the spec also says "The operations and capabilities described are in terms of measures and libraries specifically" which I believe implies that operations can be done to libraries that are not associated with measures (even though they inherently may be?) Again, going to approve this anyways but wanted to note for the future 👍
